test/boost/cdc_test.cc

```diff
@@ -849,3 +849,23 @@ SEASTAR_THREAD_TEST_CASE(test_row_delete) {
         BOOST_REQUIRE_EQUAL(*results[2].front(), data_value(cdc::operation::row_delete).serialize_nonnull());
     }, mk_cdc_test_config()).get();
 }
+
+SEASTAR_THREAD_TEST_CASE(test_mutation_without_timestamp) {
+    do_with_cql_env_thread([](cql_test_env& e) {
+        auto test = [&] (bool preimage) {
+            cquery_nofail(e, format("CREATE TABLE ks.tbl (pk int, ck int, val int, PRIMARY KEY(pk, ck)) WITH cdc = {{'enabled':true{}}}", preimage ? ",'preimage': true" : ""));
+
+            auto rows = to_bytes(*select_log(e, "tbl"));
+            BOOST_REQUIRE_EQUAL(rows.size(), 0);
+
+            cquery_nofail(e, format("DELETE FROM ks.tbl WHERE pk = 1 AND ck > 2 AND ck < 1")); // empty row range
+
+            rows = to_bytes(*select_log(e, "tbl"));
+            BOOST_REQUIRE_EQUAL(rows.size(), 0);
+
+            cquery_nofail(e, "DROP TABLE ks.tbl");
```

Comment:
shouldn't this be inside a `defer` after the CREATE?

Reply:
If anything throws before this line then the test will probably fail anyway and there's no need to clean up.